### DISCUSSION TITLE
Option for `--only-dev`?

### DISCUSSION THREAD

**Discussion by MGatner** [composer/composer#10128] [Q&A / Support] [answered]
I've searched high and low and this does not seem to exist. Is there a way to update only the packages defined in `require-dev`, the inverse of `--no-dev`? This would look like:

* `--dev` (default): Updates `require` and `require-dev`
* `--no-dev`: Updates `require`
* `--only-dev`: Updates `require-dev`

***

My instigating issue is that I have a project published with its lock file, and my development and production environments run on PHP 7.4 but the project needs to support PHP 8.x. All of the production requirements are compatible, but PHP CS Fixer needs `symfony/service-contracts` upstream which is locked to PHP 7 *or* 8. If I could update the development requirements only in my CI/CD I could still test the lock file's production capability. PHAR is not a solution.

EDIT: I solved my particular issue by locking `psr/container` to `^1.0` (see [this thread](https://github.com/symfony/symfony/issues/42683)) but I will leave this unanswered because I still see this being a desirable solution.

**Comment by stof** [comment, marked as answer]
Dev and non-dev dependencies are not resolved separately. The `--no-dev` flag only controls the installation step, not the update step. The lock file always takes the dev dependencies into account.

**Reply by MGatner** [reply]
Thanks, that makes sense.
